test/k8sT/Updates.go

```diff
@@ -259,6 +264,10 @@ func InstallAndValidateCiliumUpgrades(kubectl *helpers.Kubectl, oldVersion, newV
 		err = kubectl.CiliumInstall(helpers.CiliumDefaultDSPatch, helpers.CiliumConfigMapPatch)
 		ExpectWithOffset(1, err).To(BeNil(), "Cilium %q was not able to be deployed", newVersion)
 
+		By("Installing Cilium-Operator")
+		err = kubectl.CiliumOperatorInstall("head")
```

Comment:
this should not be "head", it should be `newVersion` as cilium-operator should always have the same image tag of cilium

Reply:
@aanm  I'm not agree in this case, we don't use kubectl-patch, and the image is not the best way to make upgrades.

The head (We decided to move away from master) Is the right thing to have, due you can set the cilium version or the head of the branch where you are. 

I do not want to over-engineering the operator installation also, I wanted to keep as simple as possible.

Reply:
The way it is right now it will not perform the upgrade test on a PR that makes changes to the cilium-operator as the upgrade test is being tested against the cilium-operator based on the commit that is on master branch.

Reply:
is how all the test are running right now. So I'm not sure why that is a problem

Reply:
Which tests? AFAIK the upgrade test is running against an cilium-agent image built **with** the changes proposed in a PR that contains the changes.

If I understood it correctly, this PR, that is adding the cilium-operator to the upgrade test is running the upgrade test against a cilium-operator image **without** the changes proposed in a PR that contains the changes. This can mean, for example, if I brake cilium-operator upgrade with a changes in my PR the upgrade will not fail and it only start failing after it is merged to master.

Reply:
discussed offline.